### ISSUE TITLE
Svelte 5 component exports not recognized in mount() return type

### ISSUE THREAD

**Issue by janosh** [pheuter/svelte-check-rs#79]
## Description

When mounting a Svelte 5 component that uses `export { prop }` to expose internal state, `svelte-check-rs` types the return value of `mount()` as `{}` instead of the component's exported interface. The original `svelte-check` correctly infers these types.

## Versions

- svelte-check-rs: 0.6.10
- svelte-check: 4.3.5
- svelte: 5.48.0

## Minimal Reproduction

**src/Counter.svelte**
```svelte
<script lang="ts">
  interface Props {
    count?: number
    name?: string
  }

  let { count = $bindable(0), name = $bindable(`World`) }: Props = $props()

  function increment() {
    count++
  }

  // Svelte 5 style exports
  export { count, name }
</script>

<button onclick={increment}>
  Hello {name}! Count: {count}
</button>
```

**tests/counter.test.ts**
```ts
import { mount, tick } from 'svelte'
import Counter from '../src/Counter.svelte'

test(`accessing exported component properties`, async () => {
  const counter = mount(Counter, {
    target: document.body,
    props: { count: 5, name: `Test` },
  })

  // These all report: Property 'X' does not exist on type '{}'
  expect(counter.count).toBe(5)
  expect(counter.name).toBe(`Test`)

  counter.count = 10
  await tick()
  expect(counter.count).toBe(10)
})
```

## Expected Behavior

No type errors - the exported properties `count` and `name` should be recognized on the mounted component instance.

## Actual Behavior

```
tests/counter.test.ts:16:18
Error: Property 'count' does not exist on type '{}'. (ts(TS2339))

tests/counter.test.ts:19:18
Error: Property 'name' does not exist on type '{}'. (ts(TS2339))

tests/counter.test.ts:22:11
Error: Property 'count' does not exist on type '{}'. (ts(TS2339))

tests/counter.test.ts:24:18
Error: Property 'count' does not exist on type '{}'. (ts(TS2339))
```

## Comparison with svelte-check

Running the original `svelte-check` on the same code reports **0 errors**.

```
$ pnpm svelte-check --tsconfig ./tsconfig.json
svelte-check found 0 errors and 0 warnings
```

## Impact

This affects any test suite that:
1. Uses Svelte's `mount()` function (from `svelte` or `@testing-library/svelte`)
2. Accesses exported component state/props after mounting

This is a common pattern in Svelte 5 testing where you need to verify or mutate component state programmatically.
